libraries/AP_Camera/AP_Camera_Backend.cpp

```diff
@@ -241,6 +250,133 @@ void AP_Camera_Backend::send_camera_settings(mavlink_channel_t chan) const
         NaN);               // focusLevel float, percentage from 0 to 100, NaN if unknown
 }
 
+// calculate poi information
+void AP_Camera_Backend::calculate_poi()
+{
+#if AP_TERRAIN_AVAILABLE
+    if (!poi_calculation.sem.take_nonblocking()) {
```

Comment:
We should use WITH_SEMAPHORE()

Reply:
add a "while(true) {" right at the top.

Reply:
check if the req_id or req_ms has been updated.  If not continue and/or sleep for 100ms.

Reply:
Why did you resolve this?  You didn't actually do what Randy requested.